middlewares/checkAuth.js

```diff
@@ -0,0 +1,19 @@
+import jwt from "jsonwebtoken";
+/** Define the function for checking the token and decoding it
+ * @params request, response, next
+ */
+
+const checkAuth = (req, res, next) => {
```

Comment:
Since we will be using different strategies, you can use ```password-jwt``` strategy to build a strong and more reliable middleware. `https://www.npmjs.com/package/passport-jwt`